packages/editor/src/pages/Editor/services/general.ts

```diff
@@ -24,27 +23,35 @@ export const fetchYaml = (url: string): Promise<{ content?: object; error?: Erro
     .then(value => ({ content: YAML.safeLoad(value) }))
     .catch(error => ({ error }));
 
-export const request = ({
+export const request = async ({
   method,
   url,
   token,
   jsonPayload,
-}: IRequest): Promise<IResponseOrError> => {
+}: IRequest): Promise<IResponseOrError & { headers?: Headers }> => {
   const headers = {
-    ...addIf(token, { Authorization: `Bearer ${token}` }),
-    ...addIf(method !== 'GET', {
+    ...getPayloadOrEmpty(token, { Authorization: `Bearer ${token}` }),
+    ...getPayloadOrEmpty(method !== 'GET', {
       'Content-Type': 'application/json; charset=utf-8',
     }),
   };
 
-  return fetch(url, {
-    method,
-    headers,
-    body: jsonPayload,
-  })
-    .then(response =>
-      response.ok ? response.json() : Promise.reject(response.statusText),
-    )
-    .then(response => ({ response }))
-    .catch(error => ({ error }));
+  try {
```

Comment:
Why change this?

Reply:
To incorporate the headers.  It turned out a fair bit easier to do this with async/await than with promises.